pkg/apis/core/types.go

```diff
@@ -2608,6 +2608,12 @@ type PodSecurityContext struct {
 	// +k8s:conversion-gen=false
 	// +optional
 	HostIPC bool
+	// Use node-wide remapped user namespace for the pods.
+	// This field is alpha-level and can be set to true on a node only if user-namespace remapping feature is enabled at runtime and kubelet.
+	// Optional: Defaults to false
+	// +k8s:conversion-gen=false
+	// +optional
+	UserNamespaceRemapping *bool
```

Comment:
shouldnt this be `HostUserNamespace *bool`?

if unset means "runtime default and matches current behavior"
if set matches the desired effect

a psp would then check that a pod can use host user namespace, or also default if unspecified to false depending on admin configuration.

Reply:
@derekwaynecarr changed it from `HostUserNamespace` to `UserNamespaceRemapping` in tha last PR update to address @liggit comment about consistency between feature-gate, api and cri.

> The mismatch between host-level, node-level, and pod-level in the feature-gate, pod API, and CRI API is hard to follow. For example, to use a hostUserNamespace field, the NodeUserNamespace feature must be enabled. another example is https://github.com/vikaschoudhary16/kubernetes/blob/userns-new/pkg/kubelet/kuberuntime/helpers.go#L281-L288 using NamespaceMode_NODE to represent host-level, and NamespaceMode_POD to represent isolated, when in fact the namespace isn't pod-isolated, but a node-wide mapping.

I am open to revert if that makes everyone happy.